notifications.py

```diff
@@ -344,6 +346,86 @@ def rule_process_groups_expiration_date(ctx):
     log.write(ctx, 'group expiration date - Finished checking research groups for reaching group expiration date | notified: {}'.format(notify_count))
 
 
+@rule.make()
+def rule_process_inactive_research_groups(ctx):
+    """Rule interface for checking for research groups that have not been modified after a certain amount of months.
+
+    :param ctx: Combined type of a callback and rei struct
+    """
+    # Only send notifications if inactivity notifications are enabled.
+    if config.inactivity_notification == 0:
+        return
+
+    # check permissions - rodsadmin only
+    if user.user_type(ctx) != 'rodsadmin':
+        log.write(ctx, "inactive research group - Insufficient permissions - should only be called by rodsadmin")
+        return
+
+    log.write(ctx, 'inactive research group - Checking Research packages for last modification dates')
+
+    zone = user.zone(ctx)
+    notify_count = 0
+    inactivity_cutoff = datetime.now() - timedelta(weeks=4.35 * config.inactivity_cutoff_months)
+    inactivity_cutoff_epoch = int((inactivity_cutoff - datetime(1970, 1, 1)).total_seconds())
+
+    # First query: obtain a list of groups with group attributes
+    iter = genquery.row_iterator(
+        "USER_GROUP_NAME",
+        "USER_TYPE = 'rodsgroup' AND USER_GROUP_NAME like 'research-%'",
+        genquery.AS_LIST, ctx
+    )
+
+    for row in iter:
+        group_name = row[0]
+        coll = '/{}/home/{}'.format(zone, group_name)
+        coll_general = '/{}/home/{}%'.format(zone, group_name)
+
+        # Second query: check if there are any files in the group, including files in the sub-folders
+        # If there are no files we do not want to notify about lack of activity
+        iter_data = genquery.row_iterator(
+            "DATA_NAME, COLL_NAME",
+            "COLL_NAME like '{}' AND USER_GROUP_NAME = '{}'".format(coll_general, group_name),
+            genquery.AS_LIST, ctx
+        )
+
+        if iter_data.total_rows() > 0:
```

Comment:
I think it would be best to add an else condition here that checks the `COLL_MODIFY_TIME` of the main collection if no data objects are currently present, so that we also send notifications for inactive groups that have no data objects anymore.

Reply:
One of the reasons for adding this notification feature was to encourage research groups to use less space. Should we still send the notification if no data objects are currently present, since in that case the group was not taking up a lot of space?

Reply:
That's a good point. I double checked with the stakeholder / data manager. He would also like to receive a notification if the group has been empty for three months. 

In that case it's more about reducing clutter in the environment, rather than clearing up storage space.